generator/ServiceClientGeneratorLib/GeneratorDriver.cs

```diff
@@ -109,6 +111,19 @@ public class GeneratorDriver
         public HashSet<string> FilesWrittenToGeneratorFolder { get; private set; }
 
         private static ConcurrentBag<string> codeGeneratedServiceNames = new ConcurrentBag<string>();
+
+        /// <summary>
+        /// Collection of services for which modeled auth resolvers (i.e. using information from their model files) should be generated.
+        /// </summary>
+        /// <remarks>
+        /// This is an allow-list for now (to prevent creating >400 files), but will be updated to a deny-list in the future (only a handful
+        /// of services such as S3 and EventBridge won't use modeled auth resolvers).
+        /// </remarks>
+        private static readonly HashSet<string> _allowListModeledAuthResolvers = new HashSet<string>
+        {
```

Comment:
I intentionally only create the resolvers for AutoScaling for now, but here's what the output looks like for other services:


<details>
<summary>CodeCatalyst (uses bearer authentication - and no region since model doesn't include SigV4)</summary>

```csharp
public class AmazonCodeCatalystAuthSchemeHandler : BaseAuthResolverHandler
{
    protected override List<IAuthSchemeOption> ResolveAuthOptions(IRequestContext requestContext)
    {
        var mappedParameters = new AmazonCodeCatalystAuthSchemeParameters
        {
            Operation = requestContext.Request.RequestName,
        };

        return AuthSchemeResolver.ResolveAuthScheme(mappedParameters);
    }
}

public class AmazonCodeCatalystAuthSchemeResolver : IAuthSchemeResolver<AmazonCodeCatalystAuthSchemeParameters>
{
    public List<IAuthSchemeOption> ResolveAuthScheme(AmazonCodeCatalystAuthSchemeParameters authParameters)
    {
        var options = new List<IAuthSchemeOption>();

        switch (authParameters.Operation)
        {
            default:
                // Default for the service, applies to all remaining operations.
                options.Add(new AuthSchemeOption { SchemeId = "smithy.api#httpBearerAuth" });
                break;
        }

        return options;
    }
}
```
</details>

<details>
<summary>CognitoIdentity (has operations that don't require auth)</summary>

```csharp
public class AmazonCognitoIdentityAuthSchemeResolver : IAuthSchemeResolver<AmazonCognitoIdentityAuthSchemeParameters>
{
    public List<IAuthSchemeOption> ResolveAuthScheme(AmazonCognitoIdentityAuthSchemeParameters authParameters)
    {
        var options = new List<IAuthSchemeOption>();

        switch (authParameters.Operation)
        {
            case "GetCredentialsForIdentityRequest":
                options.Add(new AuthSchemeOption { SchemeId = "smithy.api#noAuth" });
                break;
            case "GetIdRequest":
                options.Add(new AuthSchemeOption { SchemeId = "smithy.api#noAuth" });
                break;
            case "GetOpenIdTokenRequest":
                options.Add(new AuthSchemeOption { SchemeId = "smithy.api#noAuth" });
                break;
            case "UnlinkIdentityRequest":
                options.Add(new AuthSchemeOption { SchemeId = "smithy.api#noAuth" });
                break;
            default:
                // Default for the service, applies to all remaining operations.
                options.Add(new AuthSchemeOption { SchemeId = "aws.auth#sigv4" });
                break;
        }

        return options;
    }
}
```
</details>